src/klr/klr.cpp

```diff
@@ -373,5 +374,25 @@ int CallFirmwareProcessMain(int argc, wchar_t* argv[])
 
 int wmain(int argc, wchar_t* argv[])
 {
+    // Check for the debug flag before doing anything else
+    for (int i = 0; i < argc; i++)
+    {
+        if (StringsEqual(argv[i], L"--debug"))
+        {
+            if (!IsDebuggerPresent())
+            {
+                ::wprintf_s(L"Waiting for the debugger to attach...\r\n");
+
+                // Do not use getchar() like in corerun because it doesn't work well with remote sessions
+                while (!IsDebuggerPresent())
+                {
+                    Sleep(250);
+                }
```

Comment:
Time limitation: no. If you use the debug flag, it can be annoying if you leave it too much and then realize that the program ran. Also, it adds unnecessary complexity.

Debug break: no again. My initial code had that but then I realized that most of the times you set a breakpoint somewhere in the code and you expect that one to be hit first. You don't want every time you run the program to get a breakpoint before the one that you set, a breakpoint that you cannot disable.